src/codegen/compile.ml

```diff
@@ -4579,10 +4661,52 @@ module MakeSerialization (Strm : Stream) = struct
 
     Buffer.add_string buf "DIDL";
     add_leb128 (List.length typs);
-    List.iter add_typ typs;
+    let offsets = List.map (fun typ ->
+      let offset = Buffer.length buf in
+      add_typ typ;
+      offset)
+      typs
+    in
     add_leb128 (List.length ts);
     List.iter add_idx ts;
-    Buffer.contents buf
+    (Buffer.contents buf,
+     offsets,
+     List.map idx ts)
+
+  let set_delayed_globals (env : E.t) (set_typtbl, set_typtbl_end, set_typtbl_size, set_typtbl_idltyps) =
+       let typdesc, offsets, idltyps = type_desc env (!(env.E.typtbl_typs)) in
+       let static_typedesc =
+         Int32.(add (Blob.vanilla_lit env typdesc)  Blob.unskewed_payload_offset)
+       in
+       let tbl =
+         let buf = Buffer.create (List.length offsets * 4) in
+         begin
+           List.iter (fun offset ->
+             Buffer.add_int32_le buf
+               Int32.(add static_typedesc (of_int(offset))))
+           offsets;
+           Buffer.contents buf
+         end
+       in
+       let static_typtbl =
+         Int32.add (Blob.vanilla_lit env tbl) Blob.unskewed_payload_offset
+       in
+       let static_idltyps =
+         let buf = Buffer.create (List.length idltyps * 4) in
+         begin
+           List.iter (fun idltyp ->
+             Buffer.add_int32_le buf idltyp)
+           idltyps;
+           Buffer.contents buf
+         end
+       in
+       let static_idltyps =
+         Int32.add (Blob.vanilla_lit env static_idltyps) Blob.unskewed_payload_offset
+       in
```

Comment:
Why are you adding this as `Blob`s? Why not add this static data to static memory directly, without a header  (with `add_static`, which conveniently takes `StaticData.t`, so you don’t have to use `StaticData.as_bytes` above)

Reply:
Ignorance, really.

Reply:
Done